openedx/core/djangoapps/content_tagging/api.py

```diff
@@ -175,6 +189,147 @@ def tag_content_object(
     return get_content_tags(object_key, taxonomy_id=taxonomy.id)
 
 
+def export_content_object_children_tags(
+    course_key_str: str,
+) -> str:
+    """
+    Generates a CSV file with the tags for all the children of a course.
+    """
+    def _get_course_children_tags(course_key: CourseKey) -> tuple[dict[str, dict[int, list[str]]], dict[int, str]]:
+        """
+        Returns a tuple with a dictionary of object tags for all blocks of a course,
+        grouping by the block id and taxonomy id; and a dictionary of taxonomy ids and names.
+
+        I.e.
+        // result
+        {
+            // Block with id block-v1:edX+DemoX+Demo_Course+type@chapter+block@chapter
+            "block-v1:edX+DemoX+Demo_Course+type@chapter+block@chapter": {
+                // ObjectTags from Taxonomy with id 1
+                "1": (
+                    "Tag1",
+                    "Tag2",
+                    ...
+                ),
+                // ObjectTags from Taxonomy with id 2
+                "2": (
+                    "Tag3",
+                    ...
+                ),
+                ...
+            },
+            // Block with id block-v1:edX+DemoX+Demo_Course+type@sequential+block@sequential
+            "block-v1:edX+DemoX+Demo_Course+type@sequential+block@sequential": {
+                // ObjectTags from Taxonomy with id 1
+                "1": (
+                    "Tag2",
+                    ...
+                ),
+                ...
+            },
+        }
+
+        // taxonomies
+        {
+            "1": "Taxonomy A",
+            "2": "Taxonomy B",
+            ...
+        }
+        """
+        block_id_prefix = str(course_key).replace("course-v1:", "block-v1:", 1)
+        block_tags_records = ObjectTag.objects.filter(object_id__startswith=block_id_prefix).all()
+
+        result: dict[str, dict[int, list[str]]] = {}
+        taxonomies: dict[int, str] = {}
+
+        for object_id, block_tags in groupby(block_tags_records, lambda x: x.object_id):
+            result[object_id] = {}
+            for taxonomy_id, taxonomy_tags in groupby(block_tags, lambda x: x.tag.taxonomy_id):
+                object_tag_list = list(taxonomy_tags)
+                result[object_id][taxonomy_id] = [
+                    # If the tag is not found (deleted or freeText), use the objecttag._name instead
+                    objecttag.tag.value if objecttag.tag else objecttag.name
```

Comment:
objecttag.name is the taxonomy name, not the tag value.

Better to just use the `ObjectTag.value` property:

```suggestion
                    objecttag.value
```

Reply:
Fixed here: 4dd027aec208f1eb9a44ce4da7c43373725087c6